### PR TITLE
chore(issues): Expand stronger typing enforcement for issues

### PR DESCRIPTION
**Pull request by mrduncan**
The goal here is to get some early eyes on this, I plan to split this up into separate PRs if we're all good with it.

I had some time on a flight and took at quick pass at what is already passing here but not opted into the stronger type checking. I'm not aware of any reason not to do this, but open to thoughts. We can't quite fully enable `sentry.issues.*` or `tests.sentry.issues.*` yet.

This also has a number of trivial improvements I noticed along the way.

### CODE REVIEW

File: src/sentry/grouping/api.py [hunk 1]
```diff
@@ -121,14 +121,14 @@ def _get_enhancements(self, project) -> str:
...
-    def _get_config_id(self, project):
+    def _get_config_id(self, project: Project):
         raise NotImplementedError
 
 
 class ProjectGroupingConfigLoader(GroupingConfigLoader):
     option_name: str  # Set in subclasses
 
-    def _get_config_id(self, project):
+    def _get_config_id(self, project: Project):
```
Comment (armenzg) [inline src/sentry/grouping/api.py:131]: You need a return type. Basically every time there's no `) -> :`

File: src/sentry/grouping/enhancer/matchers.py [hunk 2]
```diff
@@ -304,7 +304,7 @@ def __init__(self, inner: FrameMatch):
     def description(self) -> str:
         return f"[ {self.inner.description} ] |"
 
-    def _to_config_structure(self, version):
+    def _to_config_structure(self, version) -> str:
```
Comment (armenzg) [inline src/sentry/grouping/enhancer/matchers.py:307]: I'm surprised that `version` is not typed and it is not complaining.

File: pyproject.toml [hunk 3]
```diff
@@ -602,6 +637,29 @@ module = [
     "sentry_plugins.base",
     "tests.sentry.api.endpoints.issues.*",
     "tests.sentry.grouping.test_fingerprinting",
+    "tests.sentry.issues.endpoints.test_actionable_items",
```
[the remark this answers is not in the document]
Comment (mrduncan) [reply pyproject.toml:640]: Not yet unfortunately, same for `sentry.issues.*`

File: src/sentry/grouping/api.py (hunk 1 above)
Comment (mrduncan) [reply src/sentry/grouping/api.py:131]: Yeah - most of these were cases where the typing was more complex than a 30s change, this at least gets us _closer_

**Comment by mrduncan** [comment]
Closing this as I'll be pulling out pieces of it into separate PRs

**Pull request closed without merging**